nuocd/nuodb_adminquery.py

```diff
@@ -137,23 +153,39 @@ def get_admin_conn(pid):
     try:
         # only interested in nuodb process on localhost, and don't
         # want to make nuoadmin rest call unless a new process is discovered.
-        _processes = subprocess.check_output(["pgrep", "^nuodb$"])
-        pids = _processes.split()
+
+        if not query_admin:
+            _processes = subprocess.check_output(["pgrep", "^nuodb$"])
+            pids = six.ensure_text(_processes).split()
+        else:
+            # if query admin, only do so if nuodb has not been found.
+            if len(running_local_processes) == 0:
+                pids = [ -1 ]
```

Comment:
What is this special -1 pid supposed to signal?

Reply:
I will be removing it, as this is change relative to fix that is not part of python3 port.  This was accidentally included in this branch.  we don't know the pid until the query is done, and we need something in this list to do the query.

Reply:
dbutson/poll changes were removed